docker/credentials_helper.go

```diff
@@ -0,0 +1,84 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/containers/image/v5/types"
+	"github.com/docker/docker-credential-helpers/client"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func GetCredentialsFromCredsStore(registry string) (types.DockerAuthConfig, error) {
+	result := types.DockerAuthConfig{}
+
+	dirname, err := os.UserHomeDir()
+	if err != nil {
+		return result, fmt.Errorf("failed to determine home directory: %w", err)
+	}
+
+	confFilePath := filepath.Join(dirname, ".docker/config.json")
+
+	f, err := os.Open(confFilePath)
+	if err != nil {
+		return result, fmt.Errorf("failed to open docker config file: %w", err)
+	}
+	defer f.Close()
+
+	conf := struct {
+		Store string `json:"credsStore"`
+	}{}
+
+	decoder := json.NewDecoder(f)
+
+	err = decoder.Decode(&conf)
+	if err != nil {
+		return result, fmt.Errorf("failed to deserialize docker config file: %w", err)
+	}
+
+	if conf.Store == "" {
+		return result, fmt.Errorf("no store configured")
+	}
+
+	helperName := fmt.Sprintf("docker-credential-%s", conf.Store)
+	p := client.NewShellProgramFunc(helperName)
+
+	credentialsMap, err := client.List(p)
+	if err != nil {
+		return result, fmt.Errorf("failed to list credentials: %w", err)
+	}
+
+	for serverUrl, username := range credentialsMap {
+		// TODO maybe use `username` if there are multiple credentials per url
+		_ = username
```

Comment:
If I am not mistaken, you can do this, if you don't need `username`:
```suggestion
	for serverUrl := range credentialsMap {

```

Reply:
Could you please elaborate on the TODO there^ ?

Reply:
I am not sure of it but I think that user can be logged under multiple logins.

In such a case we could decide based on username included in image name. But for now we don't need it. Also the namespace part of the image name/tag doesn't need to be  login, it could be name of organization in which case it's more complicated.